src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AnalyzerAssemblyService.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies
+{
+    [Export(typeof(IAnalyzerAssemblyService))]
+    [AppliesTo(ProjectCapability.DependenciesTree)]
+    internal class AnalyzerAssemblyService : IAnalyzerAssemblyService
+    {
+        private object _guard = new object();
+        private Dictionary<string, bool> _map = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
+
+        public bool ContainsDiagnosticAnalyzers(string fullPathToAssembly)
+        {
+            lock (_guard)
+            {
+                if (!_map.TryGetValue(fullPathToAssembly, out bool containsDiagnosticAnalyzers))
+                {
+                    containsDiagnosticAnalyzers = ContainsDiagnosticAnalyzersCore(fullPathToAssembly);
+                    _map.Add(fullPathToAssembly, containsDiagnosticAnalyzers);
+                }
+
+                return containsDiagnosticAnalyzers;
+            }
+        }
+
+        private static bool ContainsDiagnosticAnalyzersCore(string path)
+        {
+            try
+            {
+                using (var assemblyStream = System.IO.File.OpenRead(path))
```

Comment:
Why does this code belong in the project system? This feels like Roslyn-specific code.